src/main/java/com/google/firebase/auth/UserRecord.java

```diff
@@ -526,20 +522,30 @@ public UpdateRequest setCustomClaims(Map<String,Object> customClaims) {
       return this;
     }
 
-    UpdateRequest setValidSince(long epochSeconds) {
-      checkValidSince(epochSeconds);
-      properties.put("validSince", epochSeconds);
+    /**
+     * Updates the provider to be linked to this user\'s account.
+     *
+     * @param userProvider provider info to be linked to this user\'s account.
+     */
+    public UpdateRequest setLinkProvider(@NonNull UserProvider userProvider) {
+      properties.put("linkProviderUserInfo", userProvider);
       return this;
     }
 
-    UpdateRequest linkProvider(@NonNull UserProvider userProvider) {
-      properties.put("linkProviderUserInfo", userProvider);
+    /**
+     * Updates the identity providers to unlink from this user\'s account.
+     *
+     * @param providerIds list of identifiers for the identity providers.
+     */
+    public UpdateRequest setDeleteProviders(List<String> providerIds) {
+      checkNotNull(providerIds);
+      properties.put("deleteProvider", providerIds);
```

Comment:
Check out line 567. (Github doesn't allow you to comment on lines that haven't been changed.) The `deleteProvider` property will be overwritten if the `phone` property is null. I think you'll need to change that to append if it already exists (or set if it does not.)